File: Testes automatizados/Selenium/Selenium/src/test/java/pages/InventoryPage.java
package pages;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.Select;
import java.util.List;
import java.util.stream.Collectors;


public class InventoryPage {
    private WebDriver driver;

    private By adicionarItem1 = By.cssSelector("[data-test='add-to-cart-sauce-labs-backpack']");
    private By adicionarItem2 = By.cssSelector("[data-test='add-to-cart-sauce-labs-bike-light']");
    private By iconeCarrinho = By.cssSelector("[data-test='shopping-cart-link']");
    private By badgeCarrinho = By.cssSelector("[data-test='shopping-cart-badge']");
    private By removerItem1 = By.cssSelector("[data-test='remove-sauce-labs-backpack']");
    private By filtro = By.cssSelector("[data-test='product-sort-container']");
    private By precoItem = By.cssSelector("[data-test='inventory-item-price']");
    private By nomeItem = By.cssSelector("[data-test='inventory-item-name']");

    public InventoryPage(WebDriver driver)
    {
        this.driver = driver;
    }

    public void clicarAdicionar1()
    {
        driver.findElement(adicionarItem1).click();
    }

    public void navegarCarrinho()
    {
        driver.findElement(iconeCarrinho).click();
    }

    public void clicarAdicionar2()
    {
        driver.findElement(adicionarItem2).click();
    }

    public void clicarRemoverItem()
    {
        driver.findElement(removerItem1).click();
    }

    public boolean isBadgeCarrinhoVisivel()
    {
        return !driver.findElements(badgeCarrinho).isEmpty();
    }

    public void selecionarFiltroPorValor(String valor) {
        Select dropdown = new Select(driver.findElement(filtro));
        dropdown.selectByValue(valor);
    }

    public List<Double> obterListaDePrecos() {
        return driver.findElements(precoItem)
                .stream()
                .map(elemento -> Double.parseDouble(elemento.getText().replace("$", "").trim()))
                .collect(Collectors.toList());
    }

    public List<String> obterListaDeNomes() {
        return driver.findElements(nomeItem)
                .stream()
                .map(elemento -> elemento.getText().trim())
                .collect(Collectors.toList());
    }

}

File: Testes automatizados/Selenium em Python com BDD/pages/inventory_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import Select

class InventoryPage:
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(self.driver, 10)
        self.FILTRO_SELECTOR = (By.CSS_SELECTOR, '[data-test="product-sort-container"]')
        self.NOMES_ITENS = (By.CSS_SELECTOR, '[data-test="inventory-item-name"]')
        self.VALORES_ITENS = (By.CSS_SELECTOR, '[data-test="inventory-item-price"]')
        self.ICONE_CARRINHO = (By.CSS_SELECTOR, '[data-test="shopping-cart-link"]')
        self.ADICIONAR_ITEM_1 = (By.CSS_SELECTOR, '[data-test="add-to-cart-sauce-labs-backpack"]')
        self.ADICIONAR_ITEM_2 = (By.CSS_SELECTOR, '[data-test="add-to-cart-sauce-labs-bike-light"]')
        self.REMOVER_ITEM_1 = (By.CSS_SELECTOR, '[data-test="remove-sauce-labs-backpack"]')
        

    def adicionar_item(self):
        self.driver.find_element(*self.ADICIONAR_ITEM_1).click()

    def adicionar_2_itens(self):
        self.driver.find_element(*self.ADICIONAR_ITEM_1).click()
        self.driver.find_element(*self.ADICIONAR_ITEM_2).click()

    def clicar_carrinho(self):
        self.driver.find_element(*self.ICONE_CARRINHO).click()


    def selecionarSort(self, filtro):
        select_element = self.wait.until(EC.visibility_of_element_located(self.FILTRO_SELECTOR))
        select_element = self.driver.find_element(*self.FILTRO_SELECTOR)
        dropdown = Select(select_element)
        dropdown.select_by_visible_text(filtro)

    def obter_nomes_dos_itens(self):
        elementos = self.driver.find_elements(*self.NOMES_ITENS)
        return [elemento.text for elemento in elementos]


    def esta_em_ordem_alfabetica(self):
        nomes = self.obter_nomes_dos_itens()
        return nomes == sorted(nomes)


    def esta_em_ordem_alfabetica_inversa(self):
        nomes = self.obter_nomes_dos_itens()
        return nomes == sorted(nomes, reverse=True)
    

    def obter_valores(self):
        elementos = self.driver.find_elements(*self.VALORES_ITENS)
        return [float(elemento.text.replace('$', '').strip()) for elemento in elementos]

    def esta_em_ordem_crescente(self):
        valores = self.obter_valores()
        return valores == sorted(valores)

    def esta_em_ordem_decrescente(self):
        valores = self.obter_valores()
        return valores == sorted(valores, reverse=True)

    def removerItem(self):
        self.driver.find_element(*self.REMOVER_ITEM_1).click()